src/main/java/org/soulwing/snmp/provider/snmp4j/UserTargetStrategy.java

```diff
@@ -91,7 +92,18 @@ else if (v3Target.getAuthType() != null) {
     return userTarget;
   }
 
+  private OctetString privPassphrase(String phrase) {
+    if(phrase != null) return new OctetString(phrase);
+    return null;
```

Comment:
Given that `UsmUser` will not accept a non-null passphrase if the protocol is null, it might be best to pass the target object and return null in the case that either the passphrase or the protocol is null.

Reply:
OK, I proposed a fix to that. Did I understood you right?